framework/src/functormaterials/FunctorSmoother.C

```diff
@@ -0,0 +1,219 @@
+                                        Moose::FV::LimiterType::CentralDifference,
+                                        true,
+                                        false,
+                                        nullptr};
+                if (face_arg.fi)
+                {
+                  average += functor_in(face_arg, t);
+                  n_faces++;
+                }
+              }
+              average /= n_faces;
+            }
+            else
+              // a conceptually simple option here would be to use the smoothed version of the
+              // ElemArg to compute all the other functor arguments. Maybe with the same smoothing
+              // on the gradient calls
+              mooseError("Face averaging smoothing has only been defined for the ElemArg functor "
+                         "argument, not for ",
+                         MooseUtils::prettyCppType(&r),
+                         ". Please contact a MOOSE developer or implement it yourself.");
+          }
+          else if (_smoothing_technique == LAYERED_AVERAGE)
+          {
+            if constexpr (std::is_same_v<const Moose::ElemArg &, decltype(r)>)
+            {
+              unsigned int n_neighbors = 0;
+              for (const auto neighbor : r_elem->neighbor_ptr_range())
+                if (neighbor)
+                {
+                  n_neighbors++;
+                  average += functor_in(Moose::ElemArg{neighbor, false}, t);
+                }
+              average /= n_neighbors;
+            }
+            else
+              mooseError("Element layered averaging smoothing has only been defined for the "
+                         "ElemArg functor argument, not for ",
+                         MooseUtils::prettyCppType(&r),
+                         ". Please contact a MOOSE developer or implement it yourself.");
+          }
+          else
+          {
+            if constexpr (std::is_same_v<const Moose::ElemArg &, decltype(r)>)
+            {
+              average += base_value / 2;
+
+              // Find the neighbor with the furthest value
+              typename Moose::ADType<T>::type extreme_value = 0;
+              typename Moose::ADType<T>::type distance = 0;
+
+              unsigned int furthest_one = 0;
+              for (const auto side_index : r_elem->side_index_range())
+              {
+                auto neighbor = r_elem->neighbor_ptr(side_index);
+                if (neighbor)
+                {
+                  auto neighbor_value = functor_in(Moose::ElemArg{neighbor, false}, t);
+                  if (std::abs(neighbor_value - base_value) > distance)
+                  {
+                    furthest_one = side_index;
+                    extreme_value = neighbor_value;
+                    distance = std::abs(neighbor_value - base_value);
+                  }
+                }
+              }
+
+              if (r_elem->n_neighbors() == 1)
```

Comment:
there are no elements in libMesh with this number of neighbors

Reply:
Note that this API won't check for null neighbors

Reply:
I dont like that much. Seems like a trap

Reply:
Take it up with @roystgnr 😄

Reply:
`class Circle : public Face {` ...